engine/src/globals.cpp

```diff
@@ -864,6 +869,174 @@ void X_clear_globals(void)
 	MCDateTimeInitialize();
 }
 
+/* ---------------------------------------------------------------- */
+
+/* Helper function for X_open_environment_variables() */
+static bool
+X_open_environment_variables_store(MCArrayRef x_array,
+                                   MCStringRef p_name_str,
+                                   MCStringRef p_value,
+                                   bool p_make_global)
+{
+	MCNewAutoNameRef t_name;
+	if (!MCNameCreate(p_name_str, &t_name))
+	{
+		return false;
+	}
+
+	/* Store the environment variable in the array of
+	 * environment variables.  Note that if there are
+	 * duplicates, we always use the *first* value found in
+	 * the environment.  This matches the behaviour of most
+	 * shells. See also bug 13622.
+	 *
+	 * The global array of environment variables is *case
+	 * sensitive*, because "path" and "PATH" are distinct.
+	 */
+	MCValueRef t_current_value;
+	if (MCArrayFetchValue(x_array, true, *t_name, t_current_value))
+	{
+		return true; /* We already have a value for this variable */
+	}
+
+	if (!MCArrayStoreValue(x_array, true, *t_name, p_value))
+	{
+		return false;
+	}
+
+	// SN-2014-06-12 [[ RefactorServer ]] We don't want to duplicate
+	// the environment variables on server, as they have been copied
+	// to $_SERVER
+#ifndef _SERVER
+	/* Create a global variable for the environment variable, but
+	 * only if the variable name is a valid token beginning with a
+	 * capital letter. */
+	unichar_t t_first = MCStringGetCharAtIndex(p_name_str, 0);
```

Comment:
Where's the 'isupper' check?

To be fair I'm not entirely clear why that check was there in the first place... Is there something wrong with using env-vars which start with a lower-case letter?